tests/create_lifecycles/main.tf

```diff
@@ -8,36 +8,49 @@ module "create_lifecycles" {
 
   acl = null
 
-  lifecycle_rules = [{
-    id                                     = "transitionRule"
-    enabled                                = "true"
-    prefix                                 = "aPrefix/"
-    tags                                   = null
-    abort_incomplete_multipart_upload_days = 7
-
-    expiration = null
-
-    transitions = [{
-      date          = null
-      days          = 30
-      storage_class = "STANDARD_IA"
-      },
-      {
+  lifecycle_rules = [
+    {
+      id     = "transitionRule"
+      status = "Enabled"
+      prefix = "aPrefix/"
+
+      abort_incomplete_multipart_upload = {
+        days_after_initiation = 7
+      }
+
+      filter = {
+        prefix                   = "aPrefix/"
+        tags                     = null
+        and                      = null
+        object_size_greater_than = null
+        object_size_less_than    = null
+      }
+
+      expiration = null
+
+      transitions = [{
         date          = null
-        days          = 90
-        storage_class = "GLACIER"
-    }]
+        days          = 30
+        storage_class = "STANDARD_IA"
+        },
+        {
+          date          = null
+          days          = 90
+          storage_class = "GLACIER"
+      }]
 
-    noncurrent_version_expiration = null
+      noncurrent_version_expiration = null
 
-    noncurrent_version_transitions = []
+      noncurrent_version_transitions = []
     },
     {
-      id                                     = "expiredObjDelMarkers"
-      enabled                                = "true"
-      prefix                                 = null
-      tags                                   = { tagFilter = "testing", tagFilter2 = "123" }
```

Comment:
It would be nice to keep an example of using tags in the lifecycle rule.

Reply:
Good point.  I added the tags filter to the second lifecycle rule in the test main.tf file.  I will test to make sure it works.